datafusion/common/src/file_options/parquet_writer.rs

```diff
@@ -43,57 +48,77 @@ impl ParquetWriterOptions {
     }
 }
 
-impl TryFrom<&TableParquetOptions> for ParquetWriterOptions {
+impl TryFrom<&TableParquetOptions> for WriterPropertiesBuilder {
     type Error = DataFusionError;
 
     fn try_from(parquet_options: &TableParquetOptions) -> Result<Self> {
-        let parquet_session_options = &parquet_options.global;
+        let ParquetOptions {
```

Comment:
The `ParquetOptions` are the configuration provided within a SQL query, and therefore are [intended for use in an easily parsible format](https://github.com/apache/datafusion/blob/06895157e7f985fc4d9b0b6298c07d92abb4cc07/datafusion/common/src/config.rs#L303-L309) (refer to the `ConfigField` trait and associated macros in the linked file). 

The sorting_columns may lend itself to this use case, of being provided within a SQL query and being easier to parse. However, the same is not true for the user-provided kv_metadata.

Reply:
I have a question regarding the use case for the WriterProperties sorting_columns. It listed in [the parquet interface](https://arrow.apache.org/rust/parquet/file/properties/struct.WriterPropertiesBuilder.html#structfield.sorting_columns); is this referring to a per-row-group applied sorting that only occurs on write? Is there a use case for datafusion, given that we already sort earlier in the batch stream?

Reply:
> I have a question regarding the use case for the WriterProperties sorting_columns. It's listed in [the parquet interface](https://arrow.apache.org/rust/parquet/file/properties/struct.WriterPropertiesBuilder.html#structfield.sorting_columns);

In theory it is supposed to be used to let readers infer information from the file. I don't know how widely it is written or used by other parquet readers/writers.

IOx stores its sort information in its own metadata, so I think setting the fields in the parquet metadata could be a separate project